pkgs/development/guile-modules/guile-gcrypt/default.nix

```diff
@@ -34,6 +29,8 @@ stdenv.mkDerivation rec {
   propagatedBuildInputs = [
     libgcrypt
   ];
+  makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
```

Comment:
I couldn't find why we want to set this or what this does. Also doCheck should default to true.

Reply:
This is from [Guile's manual](https://www.gnu.org/software/guile/manual/html_node/Environment-Variables.html#Environment-Variables):

> This is a flag that can be used to tell Guile whether or not to compile Scheme source files automatically. Starting with Guile 2.0, Scheme source files will be compiled automatically, by default.
>
> If a compiled (.go) file corresponding to a .scm file is not found or is not newer than the .scm file, the .scm file will be compiled on the fly, and the resulting .go file stored away. An advisory note will be printed on the console.
>
> Compiled files will be stored in the directory `$XDG_CACHE_HOME/guile/ccache`, where XDG_CACHE_HOME defaults to the directory `$HOME/.cache`. This directory will be created if it does not already exist.

This will also fill the build logs with warnings like in the following code block if `GUILE_AUTO_COMPILE` is set:

```
building
build flags: SHELL=/nix/store/1b9p07z77phvv2hf6gm9f28syp39f1ag-bash-5.1-p16/bin/bash
Updating ./version.texi
  MAKEINFO guile-gcrypt.info
  GEN      gcrypt/base16.go
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild
;;; WARNING: compilation of /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild failed:
;;; failed to create path for auto-compiled file "/nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild"
wrote `gcrypt/base16.go'
  GEN      gcrypt/base64.go
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild
;;; WARNING: compilation of /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild failed:
;;; failed to create path for auto-compiled file "/nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild"
wrote `gcrypt/base64.go'
  GEN      gcrypt/common.go
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild
;;; WARNING: compilation of /nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild failed:
;;; failed to create path for auto-compiled file "/nix/store/mzdhzaq8jk3an8zm12jxqr0zmqn9bfb6-guile-2.2.7/bin/guild"
wrote `gcrypt/common.go'
```

Reply:
`doCheck` doesn't default to true (I also checked `pkgs/stdenv/generic/make-derivation.nix` just to be sure). I also built the package both with `doCheck` absent and set to `true`. This part of the build log shouldn't appear without `doCheck`.

```
running tests
check flags: SHELL=/nix/store/1b9p07z77phvv2hf6gm9f28syp39f1ag-bash-5.1-p16/bin/bash GUILE_AUTO_COMPILE=0 VERBOSE=y check
make  check-TESTS
make[1]: Entering directory '/build/source'
make[2]: Entering directory '/build/source'
PASS: tests/base16.scm
PASS: tests/base64.scm
PASS: tests/hash.scm
PASS: tests/hmac.scm
PASS: tests/mac.scm
PASS: tests/pk-crypto.scm
============================================================================
Testsuite summary for Guile-Gcrypt 0.3.0
============================================================================
# TOTAL: 67
# PASS:  67
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

Reply:
That read to me like we want to add ``export HOME=$(mktemp -d)`` rather than disabling compiling the schemes or am I missing something else?

and fair point with the tests. Since all the PRs handle a very similar topic I would prefer to combine them all into one. Preferable this one.

Reply:
We don't need to do that. Guile auto-compiling is intended for developers to make use of a global cache of compiled modules to speed up all throughout their Guile projects in their home directory. It's mainly used for interactive usage more than anything else.

Also, we're already compiling the schemes that are properly placed in the derivation output (`$out/lib/guile/2.2/site-ccache`) so the auto-compiling option is just getting in the way (or rather, a slight annoyance with the warnings it'll print in the build log).